lab2/kursinis.tex

```diff
@@ -47,10 +47,29 @@ \section{Context}
+    \item \textbf{Note Sharing and Interaction}: Users can interact with notes shared by others by viewing, ranking, and engaging in discussions within the topic's section. The system allows users to rank notes based on usefulness to promote the most valuable ones. Users also have options to export notes to a PDF section, download them to their device, or save them as drafts for further editing. 
+    \item \textbf{Pomodoro Timer System}: The system features a Pomodoro timer system designed to enhance productivity. Users can customize their study and break times according to their preferences. The timer notifies users when it's time to take a break, helping them maintain focus and manage their study sessions effectively.
+    \item \textbf{Goal Setting}: Users can set study goals within the system, specifying desired subjects and study times. This feature enables users to plan and organize their study sessions more efficiently, helping them stay on track with their academic or learning objectives.
+    \item \textbf{Integration with GPT API}: The project integrates with Chat GPT, providing users with additional functionalities. This integration allows the system to store user chat history and respond to user inquiries using the capabilities of Chat GPT. Users can leverage this feature to seek assistance, clarify doubts, or engage in interactive conversations within the system.
+    \end{itemize}
+
+\subsection{Tech stack and services}
+
+The EduPal system was created using a modern technology stack to support its functionalities. The basic structure of the system:
+
+\begin{itemize}
+    \item \textbf{Frontend}: The frontend of the system is developed using the React framework and JavaScript. Main component libraries were ej2-react-dropdowns, which provides UI components and specifically refers to dropdown components in React, and react-pdf, which allows the display of PDF documents. Frontend allows communication with server via fetch function which sends the HTTP request to the particular API endpoint in the backend. After the backend sends HTTP response to the frontend, React components are updated accordingly, mainly thorugh setting component state or triggering UI updates.
+    \item \textbf{Backend}: The system is built using ASP.NET Core and uses Entity Framework Core which helps following monolithic architecture, where backend is tightly integrated with frontend and database. The backend is built using .NET 7.0 and follows a RESTful API architecture, enabling it to handle HTTP requests and responses using standard methods such as GET, POST, PUT, and DELETE, with data exchanged in JSON format.Additionally, WebSocket communication is facilitated by SignalR and is used for ChatHub endpoint which, enables real-time communication.
```

Comment:
"where backend is tightly integrated with frontend and database" not fully true statement, remove it.
"tandard methods such as GET, POST, PUT, and DELETE, with data exchanged in JSON format" useless
"SignalR and is used for ChatHub endpoint which, enables r..." - signalr is used for comments